packages/app/src/app/hooks/useEnsProfile.ts

```diff
@@ -0,0 +1,24 @@
+import { useState, useEffect } from 'react'
+import { normalize } from 'viem/ens'
+import { useEnsAddress, useEnsAvatar, useEnsText } from 'wagmi'
+
+const useEnsProfile = ({ ensName, key }: { ensName: string; key?: string }) => {
+  const [name, setName] = useState<string>('')
```

Comment:
I think you could simplify the code a little bit here. Assuming the incoming `ensName` is already managed at the parent component, I'm not sure if you need to manage state and the useEffect here.

Just input the normalized name into the wagmi handlers. Or what do you think?

Reply:
Oh yeah that makes sense i will take out the useEffect and useState

Reply:
Ok now i remember. At first i just did ``` const normalizedName = normalize(ensName) ``` but i would get a client side error when the '.' button was hit. So i tried to do ```let normalizedName;``` and wrapped normalize() in a try/catch block. Then the hook wasn't updating when ensName would update, so i added the useEffect.

Considering the first approach i tried didn't work do you have any recommendations for any other techniques?

Reply:
Alright got it working

Reply:
Nice! Will try to test later today, but looks good!